pkg/syncer/cnsoperator/controller/cnsregistervolume/cnsregistervolume_controller.go

```diff
@@ -278,6 +278,23 @@ func (r *ReconcileCnsRegisterVolume) Reconcile(ctx context.Context,
 		return reconcile.Result{RequeueAfter: timeout}, nil
 	}
 
+	if instance.Spec.DiskURLPath != "" {
+		// if CNS Register Volume Instance is created with diskURLPath,
+		// confirm using CNS Volume ID, if another PV is already present with same volume ID
+		// If yes then fail.
+		pvName, found := commonco.ContainerOrchestratorUtility.GetPVNameFromCSIVolumeID(volInfo.VolumeID.Id)
+		if found {
+			if pvName != staticPvNamePrefix+volInfo.VolumeID.Id {
+				msg := fmt.Sprintf("PV: %q with the volume ID: %q for volume path: %q"+
+					"is already present. Can not create multiple PV with same disk.", pvName, volInfo.VolumeID.Id,
+					instance.Spec.DiskURLPath)
+				log.Errorf(msg)
+				setInstanceError(ctx, r, instance, msg)
+				return reconcile.Result{RequeueAfter: timeout}, nil
```

Comment:
Do we want to retry still?

Reply:
Yes, so later if PV is deleted, reconcile should succeed